### PR TITLE
Add traverse option for deeply nested mbean values

### PR DESCRIPTION
**Pull request by mreinhardt**
I decided to make this easier for the user who can just give a list of keys instead of having to write a function to apply.

### CODE REVIEW

File: examples/riemann.config [hunk 1]
```diff
@@ -10,8 +10,14 @@
...
 )
 
-(mbeans/instrumentation {:interval 10 :mbeans [{:mbean "java.lang:type=Memory" :property :HeapMemoryUsage :attribute :used}
-                                               {:mbean "java.lang:type=Runtime" :property :Uptime}]})
+(mbeans/instrumentation
+  {:interval 10
+   :mbeans [{:mbean "java.lang:type=Memory" :property :HeapMemoryUsage :attribute :used}
+            {:mbean "java.lang:type=Runtime" :property :Uptime}
+            {:mbean "java.lang:type=GarbageCollector,name=ParNew"
+             :traverse '(:LastGcInfo :memoryUsageAfterGc (keyword "Code Cache") :value :used)}
```
Comment (faxm0dem) [inline examples/riemann.config]: passing a quoted list doesn't work for me when using `keyword` in the traversal:

```
;:traverse '(:LastGcInfo :memoryUsageAfterGc (keyword "PS Eden Space") :value :used)
2016-03-24T09:25:51.000Z java.lang:type=GarbageCollector,name=PS Scavenge:LastGcInfo:memoryUsageAfterGc(keyword "PS Eden Space"):value:used N/A
```

Passing a vector instead works:

```
;:traverse [:LastGcInfo :memoryUsageAfterGc (keyword "PS Eden Space") :value :used]
2016-03-24T09:27:10.000Z java.lang:type=GarbageCollector,name=PS Scavenge:LastGcInfo:memoryUsageAfterGc:PS Eden Space:value:used 0
```

Comment (mreinhardt) [inline examples/riemann.config]: Ah, I had caught that too but forgot to update the docs. Update incoming, overwriting this commit.

**Comment by mreinhardt** [comment]
@faxm0dem Fixed docs.

**Comment by faxm0dem** [comment]
Thanks!

**Pull request merged**